lib/k9policy.ts

```diff
@@ -18,7 +18,8 @@ export enum AccessCapability {
 }
 
 export interface AccessSpec {
-    accessCapability: AccessCapability
+    accessCapability?: AccessCapability
+    accessCapabilities?: Set<AccessCapability>
```

Comment:
I wonder if you could simplify this interface by doing:
```
accessCapabilities?: Set<AccessCapability> | AccessCapability
```

I could go either way, but thinking about other styles/patterns I've seen.

Reply:
Thanks for your feedback!

I started with a single `accessCapability` or `-ilities` (preferred) attribute.

But TypeScript erases all type information at runtime and so you can't do e.g. `accessSpec.accessCapabilities instanceof AccessCapability` to dispatch to the right code:
https://stackoverflow.com/questions/33800497/check-if-an-object-implements-an-interface-at-runtime-with-typescript

So a single attribute that accepts multiple types results in 'consumers' having to detect the object type by looking for properties like a `Set`'s `entries`.  I went down that path and started to become concerned that we'd need `isAccessCapability` and `isAccessCapabilities` helper functions to do this correctly.  Which is probably fine for k9-cdk.  But what about client code?  They're going to have to handle this complecting, too.  So I went down what looked like the simplest path.

That said, having both an `accessCapability` and `accessCapabilities` attribute feels wrong.

Thinking of deprecating the single `accessCapability` attribute and removing it after people have a chance to transition to `accessCapabilities`.

wdyt?

Reply:
I've seen this done where as an input you allow an array input or a singular input but when you return the data, it's always an array. The singular input is just there for convenience as a client consumes the api. 

I'll see if I can find other examples as a reference.

Reply:
Basically like the accepted answer here https://stackoverflow.com/questions/35749833/typescript-function-taking-one-or-array-of-objects/35750406

This operates on an array instead of a Set. Not sure that really matters though.